src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/identifiers/CROIdentifier.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.digital.hmpps.personrecord.model.identifiers
+
+import org.apache.commons.lang3.builder.EqualsBuilder
+
+class CROIdentifier(inputCroId: String, inputFingerprint: Boolean, invalidInputCro: String = "") {
+
+  val croId: String = inputCroId
+  val fingerprint: Boolean = inputFingerprint
+
+  val valid: Boolean
+    get() = croId.isNotEmpty()
+
+  val invalidCro: String = invalidInputCro
+
+  override fun equals(other: Any?): Boolean {
+    return EqualsBuilder.reflectionEquals(this, other)
+  }
+
+  override fun hashCode(): Int {
+    return croId.hashCode()
+  }
+
+  override fun toString(): String {
+    return croId
+  }
+
+  companion object {
+    private const val EMPTY_CRO = ""
+    private const val SLASH = "/"
+    private const val LAST_CHARACTER = 9
+    private const val SERIAL_NUM_LENGTH = 6
+    private const val VALID_LETTERS = "ZABCDEFGHJKLMNPQRTUVWXY"
+
+    private val SF_CRO_REGEX = Regex("^SF\\d{2}/\\d{1,6}[A-Z]\$")
+    private val CRO_REGEX = Regex("^\\d{1,6}/\\d{2}[A-Z]\$")
+
+    fun from(inputCroId: String? = EMPTY_CRO): CROIdentifier {
+      if (inputCroId.isNullOrEmpty()) {
+        return CROIdentifier(EMPTY_CRO, false)
+      }
+      return toCanonicalForm(inputCroId)
```

Comment:
minor point but I quite like the ```return when``` syntax you have used elsewhere in this class, might suit here too.